05-html-css/css-grid-intro.md

```diff
@@ -0,0 +1,334 @@
+
+  1. There is one large container for these elements, and it is called **the grid container**. We designate it as the grid container by giving it the CSS rule `display: grid;`
+  1. The "many elements" that we are trying to put in a layout are called the grid items. The grid items **must** be and will only be the **direct children** of the grid container
+      - these grid items may have children inside of them, but will not be affected by this grid container
+  1. We can define the "rules" for the grid system in many ways
+      - The grid container may have CSS that determine a "template" for how all the grid items span across rows and columns
+      - The grid item may have CSS applied to it that determines how it specifically spans across rows and columns
+
+  Note: The grid items of a specific grid container do not have to be all the same kind of element.
+
+### Counting on CSS Grid
+
+CSS Grid does not begin with index 0, and instead count starting at 1.
+
+![Grid Image](./imgs/grid_lines.png)
+
+### Practice
+
+Let's look at a slightly abstract example to practice counting columns and rows and thinking in a grid.
+
+Each rectangular block of color represents one grid item. Answer the following questions:
+
+![Abstract CSS Grid Layout](./imgs/css-grid-abstract-example.png)
+
+- How many columns are in the layout total? How many are defined on the _grid container_?
+  - Are all columns equally sized? In CSS Grid, not every column needs to be the same width
+- Are the rows are in the layout total? How many are defined on the _grid container_?
+  - Are all rows equally sized? In CSS Grid, not every row needs to be the same height
+- For three different grid items, answer the following:
+  - What column line # does this grid item start at?
+  - What column line # does this grid item end at?
+  - How many columns does this grid item span?
+  - What row line # does this grid item start at?
+  - What row line # does this grid item end at?
+  - How many rows does this grid item span?
+
+### Observe
+
+Let's look at a more realistic example. Let's say we've received this wireframe for a website to create. What is the grid container? What are the rows? What are the columns? Are all columns equally sized? Are all rows equally sized? What is each grid item, and how many rows and columns do each one span?
+
+![Realistic CSS Grid Layout](./imgs/css-grid-realistic-example.png)
+
+Let's take a look at one way we'd likely make write this HTML, and its accompanying CSS to set it up to use CSS Grid.
+
+```html
+<!-- index.html -->
+<!-- stuff like the opening html tag, head tag, link to stylesheet, opening body tag -->
+
+<div class="container">
+  <header class="header">
+
+  </header>
+  <nav class="nav">
+
+  </nav>
+  <div class="content">
+
+  </div>
```

Comment:
Could you use something more semantic than a `<div>` here? `<main>` might be a good choice.